packages/bundle-analyzer/src/cli.ts

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+import path from "node:path";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { createAndUploadReport } from "./index.js";
+import { red, type Options } from "@codecov/bundler-plugin-core";
+import { type BundleAnalyzerOptions } from "./options";
+
+interface Argv {
+  buildDirectories: string[];
+  dryRun?: boolean;
+  uploadToken?: string;
+  apiUrl?: string;
+  bundleName: string;
+  debug?: boolean;
+  ignorePatterns?: string[];
+  normalizeAssetsPattern?: string;
+}
+
+const argv = yargs(hideBin(process.argv))
+  .usage("Usage: $0 <build-directories> [options]")
+  .command(
+    "$0 <build-directories...>",
+    "Analyze and upload bundle report",
+    (yargs) => {
+      return yargs.positional("buildDirectories", {
+        describe: "The paths to the build directories (can be multiple)",
+        type: "string",
+        demandOption: true,
+      });
+    },
+  )
+  .options({
+    "dry-run": {
+      alias: "d",
+      type: "boolean",
+      description: "Perform a dry run without uploading",
+      default: false,
+    },
+    "upload-token": {
+      alias: "t",
+      type: "string",
+      description: "Specify the upload token for authentication",
+    },
+    "api-url": {
+      alias: "u",
+      type: "string",
+      description: "Set the API URL",
+      default: "https://api.codecov.io",
+    },
+    "bundle-name": {
+      alias: "n",
+      type: "string",
+      description: "Set the bundle identifier in Codecov",
+      demandOption: true,
+    },
+    debug: {
+      alias: "v",
+      type: "boolean",
+      description: "Enable debug mode for additional logging",
+      default: false,
+    },
+    "ignore-patterns": {
+      alias: "i",
+      type: "array",
+      description: "Specify file patterns to ignore during the analysis",
+    },
+    "normalize-assets-pattern": {
+      alias: "p",
+      type: "string",
+      description: "Pattern to normalize asset names, e.g., '[name]-[hash].js'",
+    },
+  })
```

Comment:
So sorry for this, but I totally missed that we need to provide the ability for user to override the upload values. We need to do this in case we don't support the CI environment or they want to customize what is being sent.